src/middleware/sapCfMtlsValidation.ts

```diff
@@ -0,0 +1,137 @@
+import { FastifyRequest, HookHandlerDoneFunction } from "fastify";
+import { UnauthorizedError } from "src/model/error/UnauthorizedError.js";
+import { tokenizeDn, dnTokensMatch } from "src/util/certificateHelpers.js";
+import { log } from "src/util/logger.js";
+import {
+  CERT_ISSUER_DN_HEADER,
+  CERT_SUBJECT_DN_HEADER,
+  CERT_ROOT_CA_DN_HEADER,
+  CERT_XFCC_HEADER,
+  CERT_CLIENT_HEADER,
+  CERT_CLIENT_VERIFY_HEADER,
+} from "../constant.js";
+
+export interface MtlsValidationOptions {
+  trustedCerts: { issuer: string; subject: string }[];
+  trustedRootCaDns: string[];
+}
+
+/**
+ * Checks if the request has valid XFCC (X-Forwarded-Client-Cert) headers
+ * indicating the proxy has already verified the client certificate.
+ *
+ * Conditions:
+ * - X-Forwarded-Client-Cert header exists
+ * - X-Ssl-Client header equals "1"
+ * - X-Ssl-Client-Verify header equals "0" (verification success)
+ */
+function isXfccProxyVerified(request: FastifyRequest): boolean {
+  const xfcc = request.headers[CERT_XFCC_HEADER];
+  const sslClient = request.headers[CERT_CLIENT_HEADER];
+  const sslVerify = request.headers[CERT_CLIENT_VERIFY_HEADER];
+
+  // Handle array headers by taking first value
+  const sslClientValue = Array.isArray(sslClient) ? sslClient[0] : sslClient;
+  const sslVerifyValue = Array.isArray(sslVerify) ? sslVerify[0] : sslVerify;
+
+  return xfcc !== undefined && sslClientValue === "1" && sslVerifyValue === "0";
+}
+
+/**
+ * Creates a Fastify authentication hook for SAP Cloud Foundry mTLS validation
+ * Validates the client certificate issuer and subject as a pair, and root CA from headers
+ * Headers are always base64 encoded
+ */
+export function createSapCfMtlsValidator(options: MtlsValidationOptions) {
+  const { trustedCerts, trustedRootCaDns } = options;
+
+  if (trustedCerts.length === 0) {
+    throw new Error("mTLS validation requires at least one trusted certificate (issuer/subject pair)");
+  }
```

Comment:
Let's remove this to allow wildcard validation just by RootCA.

Minimal required configuration for `CF_MTLS_TRUSTED_CERTS` is to have property `rootCaDn` with an array length > 0, everything else is optional.